packages/suite/src/utils/suite/__tests__/parseUri.test.ts

```diff
@@ -41,4 +41,74 @@ describe('parseUri', () => {
             });
         });
     });
+
+    describe('parseUri multi', () => {
+        it('should work with any type of URIs', () => {
+            expect(parseUri('http://www.trezor.io')).toEqual({
+                address: '//www.trezor.io',
+            });
+            expect(parseUri('www.trezor.io')).toEqual({
+                address: 'www.trezor.io',
+            });
+            expect(parseUri('www.trezor.io/route')).toEqual({
+                address: 'www.trezor.io/route',
+            });
+            expect(parseUri('www.trezor.io/route?query=1&odd')).toEqual({
+                address: 'www.trezor.io/route',
+                query: '1',
+            });
+            expect(parseUri('www.trezor.io?query=1&amount=1')).toEqual({
+                address: 'www.trezor.io',
+                query: '1',
+                amount: '1',
+            });
+        });
+    });
+
+    describe('getProtocolInfo', () => {
+        it('should parse Bitcoin URI when address and amount are both available', () => {
+            const protocolInfo = getProtocolInfo(
+                'bitcoin:3QmuBaZrJNCxc5Xs7aGzZUK8RirUT8jRKf?amount=0.1',
+            );
+            expect(protocolInfo).toEqual({
+                scheme: 'bitcoin',
+                amount: 0.1,
+                address: '3QmuBaZrJNCxc5Xs7aGzZUK8RirUT8jRKf',
+            });
+        });
+
+        it('should parse Bitcoin URI when it contains not only address and amount but also valid but unsupported label and message', () => {
+            const protocolInfo = getProtocolInfo(
+                'bitcoin:3QmuBaZrJNCxc5Xs7aGzZUK8RirUT8jRKf?amount=1&label=Bender&message=Bite%20my%20shiny%20metal%20Bitcoin',
+            );
+            expect(protocolInfo).toEqual({
+                scheme: 'bitcoin',
+                amount: 1,
+                address: '3QmuBaZrJNCxc5Xs7aGzZUK8RirUT8jRKf',
+            });
+        });
+
+        it('should parse Bitcoin URI when it contains not only address and amount but also not yet existing variable', () => {
+            const protocolInfo = getProtocolInfo(
+                'bitcoin:3QmuBaZrJNCxc5Xs7aGzZUK8RirUT8jRKf?amount=1&layer=lightning',
+            );
+            expect(protocolInfo).toEqual({
+                scheme: 'bitcoin',
+                amount: 1,
+                address: '3QmuBaZrJNCxc5Xs7aGzZUK8RirUT8jRKf',
+            });
+        });
+
+        it('should throw an error when amount is missing in Bitcoin URI ', () => {
```

Comment:
please add a case for invalid amount value...
something like `bitcoin:3QmuBaZrJNCxc5Xs7aGzZUK8RirUT8jRKf?amount=hundred` and `bitcoin:3QmuBaZrJNCxc5Xs7aGzZUK8RirUT8jRKf?amount=100B`...

Reply:
5c360ef8e101cd068b4595e24c01b3a4d0ffb26d

Reply:
never mind, didn't know that parseFloat will parse it anyway...